### PR TITLE
[RFC] Update version.c to 7.4.1537

### PR DESCRIPTION
**Pull request by Shougo**
Update version.c to 7.4.1537

NA patches:

238, 244 SMACK support
1220 Tiny build
1221 configure
1222 Tiny build
1224 Makefile
1225 old style functions
1226 GRESOURCE_HDR
1227 compiler warning
1240, 1241 Makefile
1242 FEAT_EVAL
1243 Compiler warning
1245, 1251 Filelist
1270, 1272, 1280 if_python
1277 ifdef
1320, 1323, 1326, 1332 Makefile
1334, 1339 char_u cast for Windows
1340 Windows build
1344, 1345 Windows
1348 Windows GUI
1349 if_mzscheme
1350 RealWaitForChar()
1154, 1156, 1157, 1160, 1163-1168, 1173, 1178, 1181, 1188, 1269, 1278, 1279 json
1229-1235, 1238, 1239, 1244, 1246-1250, 1252-1258, 1260-1268, 1274-1275, 1283, channel/job
1286-1289, 1291, 1293-1295, 1297-1299, 1301-1304, 1306-1319, 1321, 1322, 1324, channel/job
1325, 1327-1331, 1333, 1335-1338, 1341-1343 channel/job
1351, 1353, 1355-1362, 1369-1374, 1376 channel/job
1354 Makefile
1363 Tiny Build
1364, 1368, 1375 Remove Win16 code
1367 json
1387 Win16
1377-1382, 1385, 1386, 1389, 1393, 1395, 1398, 1404 channel
1383 GVimExt
1390 configure
1391 Compiler warning
1392 Makefile
1399 MS-DOS
1400 if_perl
1402, 1409 GTK-3 support
1403 ifdef Quickfix
1407, 1408 JSON
1411 wrong indent
1412 wrong indent
1413, 1418, 1421-1423 channel
1414, 1415 Appveyor
1416 char_u fix
1417 Filelist
1419 --not-a-terminal
1420, 1424 Makefile
1425 MS-DOS

1426, 1435, 1438, 1441, 1447, 1449 channel
1427 C89
1428 DirectWrite
1429 Win32 GUI
1430, 1434, 1439, 1444-1446, 1448, 1450 JSON
1431 if_scope
1432 GUI GTK
1433 if_sniff
1436 Filelist
1437 isnan(), isinf()
1440 Windows build
1442 if_ole
1443 GTK3

1451, 1452, 1454, 1457, 1459, 1461, 1465, 1466, 1469-1472 channel
1453 Makefile
1455 JSON
1460, 1462 functions prototypes
1463 configure
1467 FEAT_FLOAT
1473, 1474 ifdef
1475 push_raw_key()

1476 UNUSED
1481 ifdef
1482-1485, 1493, 1496, 1501-1507, 1509, 1510, 1512, 1514, 1515, 1517-1520, 1522-1524 channel
1487 isinf()
1488 add_to_input_buf_csi()
1489 MSVC
1490, 1497, 1525 GUI GTK
1495 Compiler Warning
1498 JSON
1508 Makefile

1526, 1527, 1529-1532, 1534, 1536, 1537 channel

Included patches:
1405 fix the flick in complete()
1500 from neovim
1511
1757 #4509

Edit: Updated to 7.4.1376
Edit2: Updated to 7.4.1412
Edit3: Updated to 7.4.1425
Edit4: Updated to 7.4.1537
Edit5: Test patches are not N/A

### CODE REVIEW

**Comment by Shougo** [comment]
I have updated NA pathces in json 1100 patches.
And I have removed NA commas.

**Comment by justinmk** [comment]
That's a lot of NA!

> And I have removed NA commas.

Thanks ;)

**Comment by Shougo** [comment]
I have fixed the conflict and updated to 7.4.1376.

**Comment by Shougo** [comment]
I have updated to 7.4.1412.

**Comment by fmoralesc** [comment]
LGTM, will merge in a few if nobody objects.

**Comment by justinmk** [comment]
Last time we did this I found a few extras and a couple false positives. Give me some time to go over this, there's no rush :)

**Comment by justinmk** [comment]
7.4.1419-1420 (not-a-term hack) is NA.

**Comment by Shougo** [comment]
Thanks.  I have updated to 7.4.1425.

**Comment by Shougo** [comment]
I have fixed the conflict and updated to 7.4.1537.

File: src/nvim/version.c [hunk 1]
```diff
@@ -69,9 +69,324 @@ static char *features[] = {
...
+  // 1434 NA
+  // 1433 NA
+  // 1432 NA
+  // 1431 NA
+  // 1430 NA
+  // 1429 NA
+  // 1428 NA
+  // 1427 NA
+  // 1426 NA
+  // 1425 NA
```
Comment (oni-link) [inline src/nvim/version.c]: Perhaps not `NA`? We still have the dos help pages.

File: src/nvim/version.c [hunk 2]
```diff
@@ -69,9 +69,324 @@ static char *features[] = {
...
+  // 1419 NA
+  // 1418 NA
+  // 1417 NA
+  // 1416 NA
+  // 1415 NA
+  // 1414 NA
+  // 1413 NA
+  // 1412 NA
+  // 1411 NA
+  // 1410,
```
Comment (oni-link) [inline src/nvim/version.c]: This is a commit in #4064, so patch can be marked as included.

File: src/nvim/version.c [hunk 3]
```diff
@@ -69,9 +69,324 @@ static char *features[] = {
...
+  // 1409 NA
+  // 1408 NA
+  // 1407 NA
+  // 1406,
```
Comment (oni-link) [inline src/nvim/version.c]: This is a commit in #4064, so patch can be marked as included.

File: src/nvim/version.c [hunk 4]
```diff
@@ -69,9 +69,324 @@ static char *features[] = {
...
+  // 1355 NA
+  // 1354 NA
+  // 1353 NA
+  // 1352 NA
+  // 1351 NA
+  // 1350 NA
+  // 1349 NA
+  // 1348 NA
+  // 1347,
+  // 1346,
```
Comment (oni-link) [inline src/nvim/version.c]: Should be fixed with #3035.

File: src/nvim/version.c [hunk 5]
```diff
@@ -69,9 +69,324 @@ static char *features[] = {
...
+  // 1318 NA
+  // 1317 NA
+  // 1316 NA
+  // 1315 NA
+  // 1314 NA
+  // 1313 NA
+  // 1312 NA
+  // 1311 NA
+  // 1310 NA
+  // 1309 NA
```
Comment (oni-link) [inline src/nvim/version.c]: Is this `NA`?.

File: src/nvim/version.c [hunk 6]
```diff
@@ -69,9 +69,324 @@ static char *features[] = {
...
+  // 1301 NA
+  // 1300,
+  // 1299 NA
+  // 1298 NA
+  // 1297 NA
+  // 1296,
+  // 1295 NA
+  // 1294 NA
+  // 1293 NA
+  // 1292,
```
Comment (oni-link) [inline src/nvim/version.c]: Has anyone seen a new compiler complain about this? Could be marked `NA`.

File: src/nvim/version.c [hunk 7]
```diff
@@ -69,9 +69,324 @@ static char *features[] = {
...
+  // 1291 NA
+  // 1290,
```
Comment (oni-link) [inline src/nvim/version.c]: Can be marked as `NA`. We don't use vim's job features.

File: src/nvim/version.c [hunk 8]
```diff
@@ -69,9 +69,324 @@ static char *features[] = {
...
+  // 1289 NA
+  // 1288 NA
+  // 1287 NA
+  // 1286 NA
+  // 1285,
+  // 1284 NA
```
Comment (oni-link) [inline src/nvim/version.c]: Not sure about this.

File: src/nvim/version.c [hunk 9]
```diff
@@ -69,9 +69,324 @@ static char *features[] = {
...
+  // 1268 NA
+  // 1267 NA
+  // 1266 NA
+  // 1265 NA
+  // 1264 NA
+  // 1263 NA
+  // 1262 NA
+  // 1261 NA
+  // 1260 NA
+  // 1259 NA
```
Comment (oni-link) [inline src/nvim/version.c]: Don't we need this patch?

File: src/nvim/version.c [hunk 10]
```diff
@@ -69,9 +69,324 @@ static char *features[] = {
...
+  // 1361 NA
+  // 1360 NA
+  // 1359 NA
+  // 1358 NA
+  // 1357 NA
+  // 1356 NA
+  // 1355 NA
+  // 1354 NA
+  // 1353 NA
+  // 1352 NA
```
Comment (oni-link) [inline src/nvim/version.c]: Not sure about this.

File: src/nvim/version.c [hunk 11]
```diff
@@ -69,9 +69,324 @@ static char *features[] = {
...
+  // 1373 NA
+  // 1372 NA
+  // 1371 NA
+  // 1370 NA
+  // 1369 NA
+  // 1368 NA
+  // 1367 NA
   1366,
-
+  // 1365 NA
```
Comment (oni-link) [inline src/nvim/version.c]: Not sure about this.

File: src/nvim/version.c (hunk 2 above)
Comment (Shougo) [inline src/nvim/version.c]: Thanks.  I will fix it.

File: src/nvim/version.c (hunk 11 above)
Comment (Shougo) [inline src/nvim/version.c]: This is runtest.vim patch.

http://ftp.vim.org/vim/patches/7.4/7.4.1365

File: src/nvim/version.c (hunk 10 above)
Comment (Shougo) [inline src/nvim/version.c]: This is runtest.vim patch.

http://ftp.vim.org/vim/patches/7.4/7.4.1352

File: src/nvim/version.c (hunk 9 above)
Comment (Shougo) [inline src/nvim/version.c]: OK, it should be included.

File: src/nvim/version.c (hunk 8 above)
Comment (Shougo) [inline src/nvim/version.c]: It should be included.

File: src/nvim/version.c (hunk 1 above)
Comment (Shougo) [inline src/nvim/version.c]: This is MS-DOS support.
It is NA.

File: src/nvim/version.c (hunk 5 above)
Comment (Shougo) [inline src/nvim/version.c]: It is runtest.vim patch.

http://ftp.vim.org/vim/patches/7.4/7.4.1309

**Comment by Shougo** [comment]
Thank you @oni-link.
I have fixed.

**Comment by Shougo** [comment]
Rebased.

**Comment by Shougo** [comment]
1425 should be included by another PR.

**Comment by Shougo** [comment]
Rebased.

**Comment by Shougo** [comment]
* Rebase and merged
* runtest.vim changes are not N/A

**Comment by Shougo** [comment]
The reviewer is needed for N/A patches.
Can you check it?

**Comment by Shougo** [comment]
I have updated NA patches 238, 244 are N/A.

**Comment by Shougo** [comment]
I have added 1753 is already included.

**Comment by Shougo** [comment]
I have fixed the conflict.

**Comment by justinmk** [comment]
Merged. Thanks @Shougo.

- Un-marked 1266
- Updated to 1600

Note: If it is easier for you, you don't need to add notes to the commit message for version.c updates.

**Comment by Shougo** [comment]
Thank you!

**Pull request closed without merging**